exporters/autoexport/noop.go

```diff
@@ -41,3 +42,11 @@ func IsNoneSpanExporter(e trace.SpanExporter) bool {
 	_, ok := e.(noop)
 	return ok
 }
+
+var noopMetricReader = metric.NewManualReader()
```

Comment:
There could be one corner case where returning a global object could produce a bug from the user perspective. Let me try to illustrate it:

```go
ctx := context.Background()
os.Setenv("OTEL_METRICS_EXPORTER", "none")

exp, _ := autoexport.NewMetricReader(ctx)
fmt.Println(exp.Shutdown(ctx))
// Output: <nil>

exp, _ = autoexport.NewMetricReader(ctx)
fmt.Println(exp.Shutdown(ctx))
// Output: reader is shutdown
```

See: https://github.com/open-telemetry/opentelemetry-go/blob/60666c554065ac4da502fe28943eea4b938ab479/sdk/metric/manual_reader.go#L76-L94

I think we should implement something more or less like (not tested):

```go
type noopMetricReader struct {
	*metric.ManualReader // this would have to be initialized with metric.NewManualReader() when creating noopMetrics 
}

// MarshalLog returns logging data.
func (*noopMetricReader) MarshalLog() interface{} {
	return struct{ Type string }{Type: "noop metrics exporter"}
}

// Shutdown is noop.
func (*noopMetricReader) Shutdown(context.Context) error {
	return nil
}

// Collect is noop.
func (*noopMetricReader) Collect(context.Context, metricdata.ResourceMetrics) error {
	return nil
}
```

Reply:
Thanks for the catch!